line-bot-spring-boot/src/main/java/com/linecorp/bot/spring/boot/custom/LineMessagingClientFactory.java

```diff
@@ -0,0 +1,26 @@
+package com.linecorp.bot.spring.boot.custom;
+
+import com.linecorp.bot.client.LineMessagingClient;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Created by kimjh on 2017-04-14.
+ */
+public class LineMessagingClientFactory
+{
+    private final ConcurrentMap<String, LineMessagingClient> lineMessagingClientMap;
```

Comment:
Looks like concurrent modification did not occurs. Maybe enough to use HashMap^^